### ISSUE TITLE
How to use a DE method?

### ISSUE THREAD

**Issue by xinhongri** [facebookresearch/nevergrad#1253]
How to use a DE method? I want to use the DE for study. However, I don't know how to change some parameters of DE. My try is the code is as follows:

import nevergrad as ng
import numpy as np
def square(x):
    return sum((x) ** 2)
instrum = ng.p.Instrumentation(ng.p.Array(shape=(1,)))
MyDE = ng.optimizers.DifferentialEvolution(crossover=1.0, popsize=1000).set_name("MyDE", register=False)
optimizer = MyDE(parametrization=instrum, budget=5)
for i in range(optimizer.budget):
    print(i)
    x = optimizer.ask()
    print('x:', x)
    loss = square(*x.args, **x.kwargs)
    optimizer.tell(x, loss)
    recommendation = optimizer.provide_recommendation()
    print(recommendation.value)

However, while it does work, when I change the popsize from 1 to 1000, the results do not turn better as expected.
Could you tell me why and teach me how to modify?

**Comment by teytaud** [comment]
Maybe you want budget = 5*popsize and not budget = 5 ?
The budget is the number of function evaluations, not the number of iterations.

**Comment by xinhongri** [comment]
Sounds reasonable, 

Two more questinons:
1.
As DE method requires the popsize of population to be large, then it can generate more crossover or variation in the next iterations. If the budget is the number of function evaluations, not the number of iterations.  Is The instrum = ng.p.Instrumentation(ng.p.Array(shape=(1,))) right? How do I generate the init population as I want? for The instrum = ng.p.Instrumentation(ng.p.Array(shape=(1,))) only generate 1 of the init genertion.

2. what does F1 and F2 mean in DE?

Thanks a lot.

**Comment by teytaud** [comment]
This instrumentation corresponds to an optimization in dimension 1, right ? I am not sure you need DE in dimension 1.

https://facebookresearch.github.io/nevergrad/   shows an example in dimension 2 (using "parametrization=2" means working in dimension 2). This example is with NGOpt, but if you replace NGOpt by DE you just run DE.

DE accepts "tell not ask", so you can easily add points by yourself if you with.

For example <<   optimizer.tell(x, value) >> can be used for telling the optimizer that the point "x" has loss "value".

Our F1 and F2 are the standard F1 and F2 in DE:
- F2 high means that we use a lot the best point so far
- F1 high means that we use a lot the differences between points (the geometry of the problem, somehow).
 The code is here:
https://github.com/facebookresearch/nevergrad/blob/f0fd681320609146e116322756cee9bf2388be9c/nevergrad/optimization/differentialevolution.py#L175
